echo_server.py

```diff
@@ -21,10 +26,74 @@ def echo_server():
                 if len(response) < 32:
                     done = True
                 text_received = "{}{}".format(text_received, response)
-            conn.sendall(text_received)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
     except KeyboardInterrupt:
        server_socket.close()
 
+def response_ok(content, body_text):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: {}'.format(content)
+    body = body_text
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response = [first_line, timestamp, header, size, body, end]
```

Comment:
You don't need to have an `end` after the body, and you do need to have a blank line between any headers and the body. As it is, this is not a proper http response.  I suspect that some of your problems with png images may be related to that fact.